From d64211dbe4c630e7bc1ffd68ec977525f9b334e9 Mon Sep 17 00:00:00 2001
From: Raman Tenneti <rtenneti@google.com>
Date: Mon, 13 May 2019 11:57:09 -0700
Subject: [PATCH] AOSP/UnifiedEmail - bug fix to composing messages.

As part of porting https://critique.corp.google.com/#review/247540041,
needed to make EXTRA_TO, EXTRA_CC, EXTRA_BCC, EXTRA_SUBJECT, EXTRA_BODY
public so that they couuld be accessed from packages/apps/Email package.

Bug: 127320867

Test: manual - Ran the following tests on Pixel phone. Tested the email UI.

$ make -j 40
  -rw-r--r-- 1 rtenneti primarygroup 6375626 May  5 19:49 out/target/product/marlin/system/product/app/Email/Email.apk

$ make UnifiedEmailTests -j
  -rw-r--r-- 1 rtenneti primarygroup 311703 May  5 20:04 out/target/product/marlin/testcases/UnifiedEmailTests/arm64/UnifiedEmailTests.apk

$ make EmailTests -j
  -rw-r--r-- 1 rtenneti primarygroup 365023 May 13 14:39 out/target/product/marlin/testcases/EmailTests/arm64/EmailTests.apk

$ adb install -r out/target/product/marlin/system/product/app/Email/Email.apk
$ adb install -r out/target/product/marlin/testcases/EmailTests/arm64/EmailTests.apk
$ adb install -r out/target/product/marlin/testcases/UnifiedEmailTests/arm64/UnifiedEmailTests.apk

$ adb shell am instrument -w com.android.mail.emailtests
  Time: 3.519
  OK (157 tests)

Change-Id: I2a0e1048bac20c4346ec16d4bc8151297971fe0d
Merged-In: I2a0e1048bac20c4346ec16d4bc8151297971fe0d
(cherry picked from commit 8c19e4f8fa80ab21da9bd684f038f24cb1240011)
---
 src/com/android/mail/compose/ComposeActivity.java | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/com/android/mail/compose/ComposeActivity.java b/src/com/android/mail/compose/ComposeActivity.java
index b1bed47ee..d4da5327c 100644
--- a/src/com/android/mail/compose/ComposeActivity.java
+++ b/src/com/android/mail/compose/ComposeActivity.java
@@ -153,9 +153,9 @@
 
     private static final String MAIL_TO = "mailto";
 
-    private static final String EXTRA_SUBJECT = "subject";
+    public static final String EXTRA_SUBJECT = "subject";
 
-    private static final String EXTRA_BODY = "body";
+    public static final String EXTRA_BODY = "body";
 
     /**
      * Expected to be html formatted text.
@@ -168,9 +168,9 @@
 
     // Extra that we can get passed from other activities
     @VisibleForTesting
-    protected static final String EXTRA_TO = "to";
-    private static final String EXTRA_CC = "cc";
-    private static final String EXTRA_BCC = "bcc";
+    public static final String EXTRA_TO = "to";
+    public static final String EXTRA_CC = "cc";
+    public static final String EXTRA_BCC = "bcc";
 
     /**
      * An optional extra containing a {@link ContentValues} of values to be added to
